lib/elixir/lib/map.ex

```diff
@@ -93,11 +95,18 @@ defmodule Map do
   """
   @spec new(Enum.t, (term -> {key, value})) :: map
   def new(enumerable, transform) do
-    fun = fn el, acc ->
-      {k, v} = transform.(el)
-      put(acc, k, v)
-    end
-    Enum.reduce(enumerable, %{}, fun)
+    enumerable
+    |> Enum.to_list
```

Comment:
I would call `Enum.reduce/3` as you had before so you don't need to call :lists.reverse at the end on line 105. As it is today, you are reversing twice.